Feature bpta 301 implementation on preprocessing for data twitter

darylfung96 opened the pull request.

darylfung96 committed:
added tokenizer and contractions

darylfung96 committed:
added tokenizer and preprocessing with test files

darylfung96 committed:
added more test on twitter dataset

darylfung96 committed:
added sentence lower

darylfung96 committed:
added testing for edge cases
fix tokenizer not tokenizing properly
-> for e.g.: i would not becomes i'dn't
now tokenizer will tokenize it as i wouldn't

darylfung96 committed:
added comment

birdpersonteamai committed:
Rename prioritize_abbreviations.json to prioritized_abbreviations.json

birdpersonteamai committed:
Update preprocessing_helper.py

removed empty spaces.

Talk2Levi commented on src/data_utils/preprocessing_helper.py:
```diff
@@ -0,0 +1,31 @@
+import json
+import os
+
+ABBREVIATIONS_FILES_DIR = os.path.join(os.path.dirname(os.path.realpath(__file__)), '../..', 'Data', 'contractions')
```
I think we need to change folder name to abbreviations

darylfung96 replied:
It is done

Talk2Levi commented on src/data_utils/preprocessing_helper.py:
```diff
@@ -0,0 +1,31 @@
+PRIORITIZE_ABBREVIATIONS_DIR = os.path.join(ABBREVIATIONS_FILES_DIR, 'prioritize_abbreviations.json')
+
+with open(ABBREVIATIONS_DIR) as f:
+    CONTRACTIONS = json.load(f)
```
is this supposed to be contractions?

darylfung96 replied:
Yeap thanks for pointing that out

Talk2Levi commented on src/data_utils/preprocessing_helper.py:
```diff
@@ -0,0 +1,31 @@
+    for key, value in abbreviations.items():
+        contractions[value] = key
+
+    return contractions
```
So as the rest of contractions

darylfung96 replied:
These are contractions cause we are doing the reverse of abbreviations. We set the value as key instead of the other way around.

Talk2Levi commented on src/data_utils/tokenizer_wrapper.py:
```diff
@@ -0,0 +1,79 @@
+        """
+        self.tokenizer = tokenizer
+
+
```
space

Talk2Levi commented on src/data_utils/tokenizer_wrapper.py:
```diff
@@ -0,0 +1,79 @@
+            return self._tokenize_text(texts)
+        else:
+            return self._tokenize_array(texts)
+
```
space

Talk2Levi commented on src/data_utils/tokenizer_wrapper.py:
```diff
@@ -0,0 +1,79 @@
+        return contracted_tokens
+
+
+
```
space

Talk2Levi commented on src/test/data_utils_test/preprocessing_helper_test.py:
```diff
@@ -0,0 +1,23 @@
+        self.assertTrue(abbreviations.get("") is None)
+
+
+    def test_get_contractions(self):
```
so as the name contraction and abbreviations

darylfung96 committed:
fix naming convention

darylfung96 committed:
changed priotize to priotized for abbreviations

darylfung96 commented on src/data_utils/preprocessing_helper.py:
```diff
@@ -10,29 +10,22 @@
 def get_priortize_contractions():
     return __get_contractions(PRIORITIZE_ABBREVIATIONS)
 
-
```
@LeviIsAwesome I forgot to mention that with regards to the styling guide for Python - PEP 8, the standard way have 2 blank lines for functions outside of classes. For functions inside of classes, they only have a single line. As mention here: https://www.python.org/dev/peps/pep-0008/#blank-lines . I do suggest that we adhere to the standard PEP 8 styling rules but if you would prefer we could just have a single blank line instead them.

darylfung96 committed:
fixed tokenizer

darylfung96 committed:
transform instead of replace

darylfung96 committed:
Added detokenizer
Added test for detokenizer
Added word index dictionary loader
added word index dictionary saver
